src/js/components/Text/doc.js

```diff
@@ -22,6 +23,47 @@ export const doc = Text => {
     color: colorPropType.description(
       'A color identifier to use for the text color.',
     ),
+    margin: PropTypes.oneOfType([
+      PropTypes.oneOf(['none', ...MARGIN_SIZES]),
+      PropTypes.shape({
+        bottom: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        end: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        horizontal: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        left: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        right: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        start: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        top: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+        vertical: PropTypes.oneOfType([
+          PropTypes.oneOf(MARGIN_SIZES),
+          PropTypes.string,
+        ]),
+      }),
+      PropTypes.string,
+    ]).description(`The amount of margin around the component. For margin to be 
+      applied, Text needs to be wrapped in a containing Box. An object can be 
```

Comment:
Actually, what is driving this is that Text ends up as a `<span />` with `display: inline`. I think this means that horizontal margin will work but vertical margin will be ignored. So, it would be more accurate to include that detail.

Reply:
Yes, you're right. Let me specify the wording. This is the conclusion I found when first investigating: https://github.com/grommet/grommet/issues/4093#issuecomment-631759230